src/istio/mixerclient/referenced.cc

```diff
@@ -153,8 +153,8 @@ bool Referenced::Signature(const Attributes &attributes,
     } while (true);
   }
 
-  utils::MD5 hasher;
-
+  // First loop for exact_keys
+  // DON‘T DO Signature in this loop!
   for (std::size_t i = 0; i < exact_keys_.size(); ++i) {
```

Comment:
Can you factor out this loop pattern out to a method? There is already 3 loops (one for absence_keys, two for exact_keys) of similar pattern.